src/Unleash/Utilities/ToggleBootstrapUrlProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Unleash.Internal;
+
+namespace Unleash.Utilities
+{
+    public class ToggleBootstrapUrlProvider : IToggleBootstrapProvider
+    {
+        private readonly CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+        private readonly HttpClient client;
+        private readonly string path;
+        private readonly bool throwOnFail;
+
+        public ToggleBootstrapUrlProvider(string path, bool throwOnFail = false)
+        {
+            this.path = path;
+            client = new HttpClient();
+            this.throwOnFail = throwOnFail;
+        }
+
+        public ToggleBootstrapUrlProvider(string path, HttpClient client, bool throwOnFail = false)
+        {
+            this.path = path;
+            this.client = client;
+            this.throwOnFail = throwOnFail;
+        }
+
+        public string Read()
+        {
+            return Task.Run(() => FetchFile()).Result;
+        }
+
+        private async Task<string> FetchFile()
+        {
+            using (var request = new HttpRequestMessage(HttpMethod.Get, path))
+            {
+                using (var response = await client.SendAsync(request, cancellationTokenSource.Token).ConfigureAwait(false))
+                {
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        if (throwOnFail)
+                            throw new FetchingToggleBootstrapUrlFailedException("Failed to fetch feature toggles", response.StatusCode);
+
+                        return string.Empty;
```

Comment:
I don't know the code here well enough to say that this must happen but I've spent too much of my life hunting down exceptions that got swallowed. Is it rational/possible to log this error case here if it doesn't bubble?

Reply:
Yeah, there will be logging. It's on the backlog before considering myself done! Will make sure to include logging here as well! Do you have an opinion on throwOnFail or not? When you mentioned wanting things to fail in the fileprovider it got me thinking about this case as well

Reply:
Thank you! You're a star

I'm not sure I have a useful opinion on the throwOrFail so take this for what it's worth haha: I''m not sure I see a need for it to be honest; I don't see a usecase for allowing an exception to bubble (and potentially take down someone else's code) with an unhandled exception. But at the same time it defaults to false and if it's logged I'm happy. The only relevant comment I have is that I think it should inherit from the UnleashException base class so that consumers can catch that type and be safe. Personally I would have named this "failQuietly" but that feels like an unhelpful super nitpicky comment so I just kept quiet

Reply:
Cool, thank you!